### ISSUE TITLE
Can't recognize scalabrad when compiling server

### ISSUE THREAD

**Issue by jwenner** [martinisgroup/servers#228]
When I tried to compile the DirectEthernet server, it could not resolve the scalabrad dependency. The full traceback is below. Do I need to add anything to the PATH environment variable, or is there something else I need to do? 

```
C:\labrad-servers\servers\DirectEthernet>sbt
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; sup
port was removed in 8.0
Getting org.scala-sbt sbt 0.13.7 ...
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.
13.7/jars/sbt.jar ...
        [SUCCESSFUL ] org.scala-sbt#sbt;0.13.7!sbt.jar (2375ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/main/0
.13.7/jars/main.jar ...
        [SUCCESSFUL ] org.scala-sbt#main;0.13.7!main.jar (2124ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compil
er-interface/0.13.7/jars/compiler-interface-bin.jar ...
        [SUCCESSFUL ] org.scala-sbt#compiler-interface;0.13.7!compiler-interface
-bin.jar (1688ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compil
er-interface/0.13.7/jars/compiler-interface-src.jar ...
        [SUCCESSFUL ] org.scala-sbt#compiler-interface;0.13.7!compiler-interface
-src.jar (1546ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/precom
piled-2_8_2/0.13.7/jars/compiler-interface-bin.jar ...
        [SUCCESSFUL ] org.scala-sbt#precompiled-2_8_2;0.13.7!compiler-interface-
bin.jar (1639ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/precom
piled-2_9_2/0.13.7/jars/compiler-interface-bin.jar ...
        [SUCCESSFUL ] org.scala-sbt#precompiled-2_9_2;0.13.7!compiler-interface-
bin.jar (1642ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/precom
piled-2_9_3/0.13.7/jars/compiler-interface-bin.jar ...
        [SUCCESSFUL ] org.scala-sbt#precompiled-2_9_3;0.13.7!compiler-interface-
bin.jar (1624ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/action
s/0.13.7/jars/actions.jar ...
        [SUCCESSFUL ] org.scala-sbt#actions;0.13.7!actions.jar (1640ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/main-s
ettings/0.13.7/jars/main-settings.jar ...
        [SUCCESSFUL ] org.scala-sbt#main-settings;0.13.7!main-settings.jar (1658
ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/interf
ace/0.13.7/jars/interface.jar ...
        [SUCCESSFUL ] org.scala-sbt#interface;0.13.7!interface.jar (1498ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/io/0.1
3.7/jars/io.jar ...
        [SUCCESSFUL ] org.scala-sbt#io;0.13.7!io.jar (1611ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/ivy/0.
13.7/jars/ivy.jar ...
        [SUCCESSFUL ] org.scala-sbt#ivy;0.13.7!ivy.jar (1735ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/launch
er-interface/0.13.7/jars/launcher-interface.jar ...
        [SUCCESSFUL ] org.scala-sbt#launcher-interface;0.13.7!launcher-interface
.jar (1522ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/loggin
g/0.13.7/jars/logging.jar ...
        [SUCCESSFUL ] org.scala-sbt#logging;0.13.7!logging.jar (1608ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/logic/
0.13.7/jars/logic.jar ...
        [SUCCESSFUL ] org.scala-sbt#logic;0.13.7!logic.jar (1495ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/proces
s/0.13.7/jars/process.jar ...
        [SUCCESSFUL ] org.scala-sbt#process;0.13.7!process.jar (1517ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/run/0.
13.7/jars/run.jar ...
        [SUCCESSFUL ] org.scala-sbt#run;0.13.7!run.jar (1588ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/comman
d/0.13.7/jars/command.jar ...
        [SUCCESSFUL ] org.scala-sbt#command;0.13.7!command.jar (1716ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/classp
ath/0.13.7/jars/classpath.jar ...
        [SUCCESSFUL ] org.scala-sbt#classpath;0.13.7!classpath.jar (1574ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/comple
tion/0.13.7/jars/completion.jar ...
        [SUCCESSFUL ] org.scala-sbt#completion;0.13.7!completion.jar (1736ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/api/0.
13.7/jars/api.jar ...
        [SUCCESSFUL ] org.scala-sbt#api;0.13.7!api.jar (1700ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compil
er-integration/0.13.7/jars/compiler-integration.jar ...
        [SUCCESSFUL ] org.scala-sbt#compiler-integration;0.13.7!compiler-integra
tion.jar (1585ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compil
er-ivy-integration/0.13.7/jars/compiler-ivy-integration.jar ...
        [SUCCESSFUL ] org.scala-sbt#compiler-ivy-integration;0.13.7!compiler-ivy
-integration.jar (1499ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/relati
on/0.13.7/jars/relation.jar ...
        [SUCCESSFUL ] org.scala-sbt#relation;0.13.7!relation.jar (1544ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/task-s
ystem/0.13.7/jars/task-system.jar ...
        [SUCCESSFUL ] org.scala-sbt#task-system;0.13.7!task-system.jar (1533ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/tasks/
0.13.7/jars/tasks.jar ...
        [SUCCESSFUL ] org.scala-sbt#tasks;0.13.7!tasks.jar (1496ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/tracki
ng/0.13.7/jars/tracking.jar ...
        [SUCCESSFUL ] org.scala-sbt#tracking;0.13.7!tracking.jar (1572ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/testin
g/0.13.7/jars/testing.jar ...
        [SUCCESSFUL ] org.scala-sbt#testing;0.13.7!testing.jar (1517ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/contro
l/0.13.7/jars/control.jar ...
        [SUCCESSFUL ] org.scala-sbt#control;0.13.7!control.jar (1514ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/collec
tions/0.13.7/jars/collections.jar ...
        [SUCCESSFUL ] org.scala-sbt#collections;0.13.7!collections.jar (1713ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/increm
ental-compiler/0.13.7/jars/incremental-compiler.jar ...
        [SUCCESSFUL ] org.scala-sbt#incremental-compiler;0.13.7!incremental-comp
iler.jar (1641ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compil
e/0.13.7/jars/compile.jar ...
        [SUCCESSFUL ] org.scala-sbt#compile;0.13.7!compile.jar (1585ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/persis
t/0.13.7/jars/persist.jar ...
        [SUCCESSFUL ] org.scala-sbt#persist;0.13.7!persist.jar (1510ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/classf
ile/0.13.7/jars/classfile.jar ...
        [SUCCESSFUL ] org.scala-sbt#classfile;0.13.7!classfile.jar (1451ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/cross/
0.13.7/jars/cross.jar ...
        [SUCCESSFUL ] org.scala-sbt#cross;0.13.7!cross.jar (1468ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/cache/
0.13.7/jars/cache.jar ...
        [SUCCESSFUL ] org.scala-sbt#cache;0.13.7!cache.jar (1459ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/test-a
gent/0.13.7/jars/test-agent.jar ...
        [SUCCESSFUL ] org.scala-sbt#test-agent;0.13.7!test-agent.jar (1528ms)
downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/apply-
macro/0.13.7/jars/apply-macro.jar ...
        [SUCCESSFUL ] org.scala-sbt#apply-macro;0.13.7!apply-macro.jar (1459ms)
:: retrieving :: org.scala-sbt#boot-app
        confs: [default]
        51 artifacts copied, 0 already retrieved (15709kB/488ms)
[info] Loading project definition from C:\labrad-servers\servers\DirectEthernet\
project
[info] Updating {file:/C:/labrad-servers/servers/DirectEthernet/project/}directe
thernet-build...
[info] Resolving com.typesafe.sbteclipse#sbteclipse-plugin;3.0.0 ...
  [info] Resolving com.typesafe.sbteclipse#sbteclipse-plugin;3.0.0 ...
  [info] Resolving com.typesafe.sbteclipse#sbteclipse-core;3.0.0 ...
  [info] Resolving com.typesafe.sbteclipse#sbteclipse-core;3.0.0 ...
  [info] Resolving org.scalaz#scalaz-core_2.10;7.0.2 ...
  [info] Resolving org.scala-lang#scala-library;2.10.4 ...
  [info] Resolving org.scalaz#scalaz-effect_2.10;7.0.2 ...
  [info] Resolving org.xerial.sbt#sbt-pack;0.6.5 ...
  [info] Resolving org.fusesource.scalate#scalate-core_2.10;1.6.1 ...
  [info] Resolving org.fusesource.scalate#scalate-util_2.10;1.6.1 ...
  [info] Resolving org.slf4j#slf4j-api;1.6.1 ...
  [info] Resolving org.slf4j#slf4j-api;1.6.1 ...
  [info] Resolving org.slf4j#slf4j-parent;1.6.1 ...
  [info] Resolving org.slf4j#slf4j-parent;1.6.1 ...
  [info] Resolving org.scala-lang#scala-compiler;2.10.4 ...
  [info] Resolving org.scala-lang#scala-reflect;2.10.4 ...
  [info] Resolving org.kamranzafar#jtar;2.2 ...
  [info] Resolving org.slf4j#slf4j-nop;1.7.5 ...
  [info] Resolving org.slf4j#slf4j-nop;1.7.5 ...
  [info] Resolving org.slf4j#slf4j-parent;1.7.5 ...
  [info] Resolving org.slf4j#slf4j-api;1.7.5 ...
  [info] Resolving org.scala-sbt#sbt;0.13.7 ...
  [info] Resolving org.scala-sbt#main;0.13.7 ...
  [info] Resolving org.scala-sbt#actions;0.13.7 ...
  [info] Resolving org.scala-sbt#classpath;0.13.7 ...
  [info] Resolving org.scala-sbt#launcher-interface;0.13.7 ...
  [info] Resolving org.scala-sbt#interface;0.13.7 ...
  [info] Resolving org.scala-sbt#io;0.13.7 ...
  [info] Resolving org.scala-sbt#control;0.13.7 ...
  [info] Resolving org.scala-sbt#completion;0.13.7 ...
  [info] Resolving org.scala-sbt#collections;0.13.7 ...
  [info] Resolving jline#jline;2.11 ...
  [info] Resolving org.scala-sbt#api;0.13.7 ...
  [info] Resolving org.scala-sbt#compiler-integration;0.13.7 ...
  [info] Resolving org.scala-sbt#incremental-compiler;0.13.7 ...
  [info] Resolving org.scala-sbt#logging;0.13.7 ...
  [info] Resolving org.scala-sbt#process;0.13.7 ...
  [info] Resolving org.scala-sbt#relation;0.13.7 ...
  [info] Resolving org.scala-sbt#compile;0.13.7 ...
  [info] Resolving org.scala-sbt#persist;0.13.7 ...
  [info] Resolving org.scala-tools.sbinary#sbinary_2.10;0.4.2 ...
  [info] Resolving org.scala-sbt#classfile;0.13.7 ...
  [info] Resolving org.scala-sbt#compiler-ivy-integration;0.13.7 ...
  [info] Resolving org.scala-sbt#ivy;0.13.7 ...
  [info] Resolving org.scala-sbt#cross;0.13.7 ...
  [info] Resolving org.scala-sbt.ivy#ivy;2.3.0-sbt-fccfbd44c9f64523b61398a015578
4dcbaeae28f ...
  [info] Resolving com.jcraft#jsch;0.1.46 ...
  [info] Resolving org.json4s#json4s-native_2.10;3.2.10 ...
  [info] Resolving org.json4s#json4s-core_2.10;3.2.10 ...
  [info] Resolving org.json4s#json4s-ast_2.10;3.2.10 ...
  [info] Resolving com.thoughtworks.paranamer#paranamer;2.6 ...
  [info] Resolving org.scala-lang#scalap;2.10.0 ...
  [info] Resolving org.spire-math#jawn-parser_2.10;0.6.0 ...
  [info] Resolving org.spire-math#json4s-support_2.10;0.6.0 ...
  [info] Resolving org.scala-sbt#run;0.13.7 ...
  [info] Resolving org.scala-sbt#task-system;0.13.7 ...
  [info] Resolving org.scala-sbt#tasks;0.13.7 ...
  [info] Resolving org.scala-sbt#tracking;0.13.7 ...
  [info] Resolving org.scala-sbt#cache;0.13.7 ...
  [info] Resolving org.scala-sbt#testing;0.13.7 ...
  [info] Resolving org.scala-sbt#test-agent;0.13.7 ...
  [info] Resolving org.scala-sbt#test-interface;1.0 ...
  [info] Resolving org.scala-sbt#main-settings;0.13.7 ...
  [info] Resolving org.scala-sbt#apply-macro;0.13.7 ...
  [info] Resolving org.scala-sbt#command;0.13.7 ...
  [info] Resolving org.scala-sbt#logic;0.13.7 ...
  [info] Resolving org.scala-sbt#compiler-interface;0.13.7 ...
  [info] Resolving org.scala-sbt#precompiled-2_8_2;0.13.7 ...
  [info] Resolving org.scala-sbt#precompiled-2_9_2;0.13.7 ...
  [info] Resolving org.scala-sbt#precompiled-2_9_3;0.13.7 ...
  [info] Resolving org.scala-lang#jline;2.10.4 ...
  [info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading https://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.
sbteclipse/sbteclipse-plugin/scala_2.10/sbt_0.13/3.0.0/jars/sbteclipse-plugin.ja
r ...
[info]  [SUCCESSFUL ] com.typesafe.sbteclipse#sbteclipse-plugin;3.0.0!sbteclipse
-plugin.jar (1489ms)
[info] downloading https://repo.typesafe.com/typesafe/ivy-releases/com.typesafe.
sbteclipse/sbteclipse-core/scala_2.10/sbt_0.13/3.0.0/jars/sbteclipse-core.jar ..
.
[info]  [SUCCESSFUL ] com.typesafe.sbteclipse#sbteclipse-core;3.0.0!sbteclipse-c
ore.jar (1745ms)
[info] downloading https://repo1.maven.org/maven2/org/slf4j/slf4j-nop/1.7.5/slf4
j-nop-1.7.5.jar ...
[info]  [SUCCESSFUL ] org.slf4j#slf4j-nop;1.7.5!slf4j-nop.jar (182ms)
[info] downloading https://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.5/slf4
j-api-1.7.5.jar ...
[info]  [SUCCESSFUL ] org.slf4j#slf4j-api;1.7.5!slf4j-api.jar (192ms)
[info] Done updating.
[info] Set current project to DirectEthernet (in build file:/C:/labrad-servers/s
ervers/DirectEthernet/)
> compile
[info] Updating {file:/C:/labrad-servers/servers/DirectEthernet/}directethernet.
..
[info] Resolving org.scala-lang#scala-library;2.11.7 ...
  [info] Resolving org.labrad#scalabrad_2.11;0.4.1 ...
  [info] Resolving org.labrad#scalabrad_2.11;0.4.1 ...
[warn]  module not found: org.labrad#scalabrad_2.11;0.4.1
[warn] ==== local: tried
[warn]   C:\Users\rottmayer-user\.ivy2\local\org.labrad\scalabrad_2.11\0.4.1\ivy
s\ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/org/labrad/scalabrad_2.11/0.4.1/scalabra
d_2.11-0.4.1.pom
[info] Resolving org.pcap4j#pcap4j-core;1.5.0 ...
  [info] Resolving org.pcap4j#pcap4j-core;1.5.0 ...
  [info] Resolving org.pcap4j#pcap4j;1.5.0 ...
  [info] Resolving org.pcap4j#pcap4j;1.5.0 ...
  [info] Resolving org.sonatype.oss#oss-parent;7 ...
  [info] Resolving net.java.dev.jna#jna;4.1.0 ...
  [info] Resolving net.java.dev.jna#jna;4.1.0 ...
  [info] Resolving org.slf4j#slf4j-api;1.6.4 ...
  [info] Resolving org.slf4j#slf4j-api;1.6.4 ...
  [info] Resolving org.slf4j#slf4j-parent;1.6.4 ...
  [info] Resolving org.slf4j#slf4j-parent;1.6.4 ...
  [info] Resolving org.scalatest#scalatest_2.11;2.2.0 ...
  [info] Resolving org.scalatest#scalatest_2.11;2.2.0 ...
  [info] Resolving org.scala-lang#scala-reflect;2.11.1 ...
  [info] Resolving org.scala-lang#scala-reflect;2.11.1 ...
  [info] Resolving org.scala-lang.modules#scala-xml_2.11;1.0.1 ...
  [info] Resolving org.scala-lang.modules#scala-xml_2.11;1.0.1 ...
  [info] Resolving org.scala-lang#scala-compiler;2.11.7 ...
  [info] Resolving org.scala-lang#scala-reflect;2.11.7 ...
  [info] Resolving org.scala-lang.modules#scala-xml_2.11;1.0.4 ...
  [info] Resolving org.scala-lang.modules#scala-parser-combinators_2.11;1.0.4 ..
.
  [info] Resolving jline#jline;2.12.1 ...
[info] downloading https://repo1.maven.org/maven2/org/pcap4j/pcap4j-core/1.5.0/p
cap4j-core-1.5.0.jar ...
[info]  [SUCCESSFUL ] org.pcap4j#pcap4j-core;1.5.0!pcap4j-core.jar (421ms)
[info] downloading https://repo1.maven.org/maven2/net/java/dev/jna/jna/4.1.0/jna
-4.1.0.jar ...
[info]  [SUCCESSFUL ] net.java.dev.jna#jna;4.1.0!jna.jar (303ms)
[info] downloading https://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.6.4/slf4
j-api-1.6.4.jar ...
[info]  [SUCCESSFUL ] org.slf4j#slf4j-api;1.6.4!slf4j-api.jar (149ms)
[info] downloading https://repo1.maven.org/maven2/org/scalatest/scalatest_2.11/2
.2.0/scalatest_2.11-2.2.0.jar ...
[info]  [SUCCESSFUL ] org.scalatest#scalatest_2.11;2.2.0!scalatest_2.11.jar(bund
le) (1405ms)
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2
.11.1/scala-reflect-2.11.1.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-reflect;2.11.1!scala-reflect.jar (762
ms)
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-x
ml_2.11/1.0.1/scala-xml_2.11-1.0.1.jar ...
[info]  [SUCCESSFUL ] org.scala-lang.modules#scala-xml_2.11;1.0.1!scala-xml_2.11
.jar(bundle) (222ms)
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.labrad#scalabrad_2.11;0.4.1: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]
[warn]  Note: Unresolved dependencies path:
[warn]          org.labrad:scalabrad_2.11:0.4.1 (C:\labrad-servers\servers\Direc
tEthernet\build.sbt#L10-14)
[warn]            +- org.labrad:directethernet_2.11:1.2.0
[trace] Stack trace suppressed: run last *:update for the full output.
[error] (*:update) sbt.ResolveException: unresolved dependency: org.labrad#scala
brad_2.11;0.4.1: not found
[error] Total time: 7 s, completed Oct 5, 2015 8:20:15 PM
>
```

**Comment by DanielSank** [comment]
I don't see anything in `build.sbt` that fetches scalabrad. I wonder if there's a way to use github as a source repo in sbt.

**Comment by maffoo** [comment]
In the last change to build.sbt we took out the bintray resolver. @ejeffrey and I tested this, but it's possible we made a mistake and still had the package cached locally, so didn't realize that resolver was needed to find it. We should add that resolver line back and it will fix this issue.

@DanielSank, I would prefer to use published artifacts rather than git project references in sbt, but it's something we can explore. I have used those kind of source references in the past and had problems with them; they're not very well supported in sbt, really just sort of hacked in and not used by many people. It's something we can explore, though. I have heard that this sort of project reference is something that's being worked on for sbt 1.x, so it might get better in the near future.
